Sources/NIOSSH/Connection State Machine/Operations/AcceptsUserAuthMessages.swift

```diff
@@ -64,15 +72,26 @@ extension AcceptsUserAuthMessages {
         let result = try self.userAuthStateMachine.receiveUserAuthFailure(message)
 
         if let future = result {
-            return .possibleFutureMessage(future.map(Self.transform(_:)))
+            return .possibleFutureMessage(future.map(self.transform(_:)))
         } else {
             return .noMessage
         }
     }
 
-    private static func transform(_ result: NIOSSHUserAuthenticationResponseMessage) -> SSHMultiMessage {
+    mutating func receiveUserAuthBanner(_ message: SSHMessage.UserAuthBannerMessage) throws -> SSHConnectionStateMachine.StateMachineInboundProcessResult {
+      try self.userAuthStateMachine.receiveUserAuthBanner(message)
+      return .event(UserAuthBannerEvent(message: message.message, languageTag: message.languageTag))
+    }
+
+    private func transform(_ result: NIOSSHUserAuthenticationResponseMessage) -> SSHMultiMessage {
```

Comment:
Unfortunately we can't make this non-`static` without causing ourselves some trouble. NIO futures are "eager": if they have a value already and we attach a callback to them, that callback will be dispatched synchronously at the call site. Because the futures from which this is called are now closing over `self`, this risks violating the law of exclusivity at runtime: we're reading from `self` while engaged in a mutation of self.

Rather than make this a member function, can I propose we give it an extra argument that contains the optional `banner`? We can then close over that value as part of the callback.